### DISCUSSION TITLE
Updating line item returns no errors

### DISCUSSION THREAD

**Discussion by sstottelaar** [duncanmcclean/simple-commerce#1092] [Help] [answered]
I've been scratching my head on this one for a while. In my cart overview the user can update the quantity of a line item, this works fine. But if the product has stock and the updated amount exceeds it, nothing happens and no error is returned. The controller is intended to return an error, but that doesn't seem to work working:

```php
// Ensure there's enough stock to fulfill the customer's quantity
        if ($product->purchasableType() === ProductType::Product) {
            if (is_int($product->stock()) && $request->quantity > $product->stock()) {
                return $this->withErrors($request, __("There's not enough stock to fulfil the quantity you selected. Please try again later."));
            }
        } elseif ($product->purchasableType() === ProductType::Variant) {
            $variant = $request->has('variant')
                ? $product->variant($request->get('variant'))Hey, I’ve been scratching my head on this one for a while. In my cart overview, the user can update the quantity of a line item, and this works fine. However, if the product has stock and the updated amount exceeds it, nothing happens and no error is returned. The controller is intended to return an error, but that’s not working:
                : $product->variant($lineItem->variant()['variant']);

            if ($variant !== null && is_int($variant->stock()) && $request->quantity > $variant->stock()) {
                return $this->withErrors($request, __("There's not enough stock to fulfil the quantity you selected. Please try again later."));
            }
        }
```

This is from the CartItemController.php:144. So it seems the `return $this->withErrors` is not working correctly. Is this a known issue or do I need to do something differently?

**Comment by duncanmcclean** [comment, marked as answer]
I believe you'll need to catch and display the errors on your frontend. 

I haven't tested this but I _think_ you should be able to use [Statamic's `{{ get_errors }}` tag](https://statamic.dev/tags/get_errors#getting-all-errors-grouped-by-field). 

If that doesn't work, I'll need to take a deeper look when I have some free time.

**Reply by sstottelaar** [reply]
Was not aware of this, thanks for pointing it out!
